### PR TITLE
ATO-570: Create Authorisation lambda

### PR DESCRIPTION
**Pull request by gn-GDS**
- Add dynamoDB prefix to Client Service
- Add HTTPS egress security group

## Checklist

- [ ] Impact on orch and auth mutual dependencies has been checked.

- [ ] A UCD review has been performed.

### CODE REVIEW

File: template.yaml [hunk 1]
```diff
@@ -85,10 +103,16 @@ Mappings:
...
       docAppAud: https://www.review-b.integration.account.gov.uk
+      docAppAuthKmsKeyAlias: ""
+      docAppAuthorisationUrl: ""
+      docAppClientId: authOrchestratorDocApp
       docAppCredentialTableKeyArn: arn:aws:kms:eu-west-2:761723964695:key/37aea309-c4a2-4ac7-8361-c1789f54df77
+      docAppEncryptionKeyId: 0948190d-384c-498d-81e2-a20dd30f147c
       docAppSigningKeyArn: arn:aws:kms:eu-west-2:761723964695:key/031ece8b-a59a-4991-a2aa-98f486c69f60
       clientRegistryTableKeyArn: arn:aws:kms:eu-west-2:761723964695:key/f8897cf9-760f-463e-bb27-a86cc9e629de
       userProfileTableKeyArn: arn:aws:kms:eu-west-2:761723964695:key/7f17084d-14a7-4482-8a7b-e392d1f60d41
+      txmaEventEncryptionKeyArn: ""
```
Comment (CarlyG55) [inline template.yaml:114]: Ooh I imagine this is something we will need on other lambdas as well?

File: template.yaml [hunk 2]
```diff
@@ -857,6 +887,239 @@ Resources:
...
+      AutoPublishAlias: latest
+      CodeUri: ./oidc-api
+      Handler: uk.gov.di.authentication.oidc.lambda.AuthorisationHandler::handleRequest
+      ReservedConcurrentExecutions: 1
+      LoggingConfig:
+        LogGroup: !Ref AuthorisationFunctionLogGroup
+      VpcConfig:
+        SecurityGroupIds:
+          - !Ref LambdaSecurityGroup
+          - !Ref HttpsEgressSecurityGroup
```
Comment (CarlyG55) [inline template.yaml:907]: Do you mind making a quick change in this PR and adding this to doc app as well?

File: template.yaml [hunk 3]
```diff
@@ -857,6 +887,239 @@ Resources:
...
+  AuthorisationFunctionCrossAccountPermission:
+    Type: AWS::Lambda::Permission
+    Properties:
+      FunctionName: !Ref AuthorisationFunction.Alias
+      Action: lambda:InvokeFunction
+      Principal: apigateway.amazonaws.com
+      SourceArn: !Sub
+        - 'arn:aws:execute-api:eu-west-2:${AccountId}:${ApiId}/*/GET/authorize'
+        - AccountId: !FindInMap [ EnvironmentConfiguration, !Ref Environment, authAccountId ]
+          ApiId: !FindInMap [ EnvironmentConfiguration, !Ref Environment, authApiId ]
```
Comment (CarlyG55) [inline template.yaml:1033]: Are we going to need one of these for post as well?

File: template.yaml [hunk 4]
```diff
@@ -857,6 +887,239 @@ Resources:
             Unit: Count
   #endregion
 
+
```
Comment (isaac-GDS) [inline template.yaml:890]: Add region comment to distinguish between lambdas

**Comment by isaac-GDS** [comment]
Raising a PR for the current lambdas already in but could we start including the logs to splunk https://github.com/govuk-one-login/authentication-api/pull/4504/files

File: template.yaml (hunk 3 above)
Comment (gn-GDS) [reply template.yaml]: Correct, added permissions and ran a post journey 👍

**Review by CarlyG55 — APPROVED** [review]

**Pull request merged**